### ISSUE TITLE
Error LNK2019	unresolved external symbol, Shark.

### ISSUE THREAD

**Issue by mohammadAdnan** [Shark-ML/Shark#144]
Hello, 

When i run linearRegression tutorial, i get the following errors:

Error	LNK1120	3 unresolved externals	sharkTesting	C:\Users\Avrahim\Documents\Visual Studio 2015\Projects\Machine learning in C\sharkTesting\x64\Debug\sharkTesting.exe	1

Error	LNK2019	unresolved external symbol "void __cdecl shark::csvStringToData(..)

Error	LNK2019	unresolved external symbol "public: virtual void __cdecl shark::LDA::train(..)

Error	LNK2001	unresolved external symbol "public: virtual void __cdecl shark::LDA::train(..)

**Comment by Ulfgard** [comment]
try copying the shark library into the same folder or tell your operating system where to find the library.

https://msdn.microsoft.com/en-us/library/7d83bc18.aspx

**Comment by mohammadAdnan** [comment]
@Ulfgard  Yeah i know it, but i can't find shark.dll ?

**Comment by Ulfgard** [comment]
its probably in yout build/lib/ directory. could also be named libshark or similar. not sure how the naming convention on windows is
________________________________

**Comment by mohammadAdnan** [comment]
@Ulfgard, i got it, i did not build shark project, because i did not use Cmake GUI, but instead i did that step on cmd using the following:

cmake  
-G "Visual Studio 14 2015 Win64"
-DBOOST_INCLUDEDIR:PATH="C:\Users\Avrahim\Documents\Visual Studio 2015\Projects\Shark-3.1.0\packages\boost.1.63.0.0\lib\native\include"
-DBOOST_LIBRARYDIR:PATH="C:\Users\Avrahim\Documents\Visual Studio 2015\Projects\Shark-3.1.0\packages\boost_libs" 
-DDEBUG=OFF -DPROFILE=OFF ..

Why i did that? Because when i try to generate the project in cmake GUI, it did not work. So after i did the first step on cmd. Next ,i went to build folder opened shark.sln and clicked on ALL_BUILD, then i choose the x64 and Debug. However, it's not working! So now i don't have both dll or lib for shark.

**Comment by Ulfgard** [comment]
Hi,

"However, it's not working!"

Bad to hear. What exactly is not working?

________________________________

**Comment by mohammadAdnan** [comment]
The point is, after i have the build folder successfully for shark, i can't debug it in order to get the libraries and start developing projects under shark! I am working in visual studio 2015 and windows 10 64 bit.

**Comment by Ulfgard** [comment]
Please be more specific. It is really hard to help you otherwise.

What does "i can't debug it" mean? why do you need to debug it in order to get the libraries? If you clicked on "BUILD_ALL",  they should be right there in your build folder, probably in some path like Debug/x86_64/lib or similar. (very similar to where your own vs projects would put the files).

**Comment by mohammadAdnan** [comment]
Thanks Ulfgard,

First, i did this:

cmake
-G "Visual Studio 14 2015 Win64"
-DBOOST_INCLUDEDIR:PATH="C:\Users\Avrahim\Documents\Visual Studio 2015\Projects\Shark-3.1.0\packages\boost.1.63.0.0\lib\native\include"
-DBOOST_LIBRARYDIR:PATH="C:\Users\Avrahim\Documents\Visual Studio 2015\Projects\Shark-3.1.0\packages\boost_libs"
-DDEBUG=OFF -DPROFILE=OFF ..

Second, i went to the build folder and opened shark.sln.

Third, i clicked on build_all then i choose x64 and build, but it did not work.

**Comment by mohammadAdnan** [comment]
I got this anyway when i build shark in cmd usign cmake:

-- The C compiler identification is MSVC 19.0.24213.1
-- The CXX compiler identification is MSVC 19.0.24213.1
-- Check for working C compiler: F:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe
-- Check for working C compiler: F:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: F:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe
-- Check for working CXX compiler: F:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/x86_amd64/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Will build: Release
-- Looking for pthread.h
-- Looking for pthread.h - not found
-- Found Threads: TRUE
-- Boost version: 1.63.0
-- Found the following Boost libraries:
--   system
--   date_time
--   filesystem
--   program_options
--   serialization
--   thread
--   unit_test_framework
-- Using boost from
-- OpenMP found
-- HDF5 not found, skip
-- No usable CBLAS Library found. No fast linear Algebra used.
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    BLAS_LIBRARY
    DEBUG
    LAPACK_LIBRARY
    PROFILE

Do you see anything wrong?

**Comment by mohammadAdnan** [comment]
@Ulfgard  I hope my problem is clear to you now? It's normal to build library using cmake, but in order to get libraries i opened the build folder that is generated from cmake, then open shark.sln, but when i choose build, it don't work.

**Comment by Ulfgard** [comment]
hi,

Please take into account, that I am sitting on a different operating system with a different IDE a few 1000 miles away from you and the only thing i can work with is the information you supply. Again: what does "when i choose build, it don't work?" mean? I doubt that the error message you see is "it does not work".

If you have trouble finding the error message, just supply the whole build log generated by BUILD_ALL. If that succeeds, try to run the tutorial and if that fails, also supply me the full error message.

**Comment by mohammadAdnan** [comment]
Thanks Ulfgard, i appreciate your time and response. I will reform my question better.

After i followed the installation steps in your homepage, i got the build folder for shark; I generated shark using cmake, where i got the build folder for shark. What i should do next? I did the follwing:

Next, i need shark.dll and shark.Lib in order to link my project against shark, but i can't get .dll and .lib! You told me that i need to open shark.sln in shark/build/shark.sln and click on ALL_BUILD then click build in order to get .dll and .lib for your project, okay? But the last step is not working.
